test/modules/obsagent-client/src/main.rs

```diff
@@ -1,210 +1,43 @@
-fn delayed_interval(duration: Duration) -> impl Stream<Item = tokio::time::Instant> {
-    opentelemetry::util::tokio_interval_stream(duration).skip(1)
-}
-
-fn init_meter(period: f64, otlp_endpoint: String) -> metrics::Result<PushController> {
-    let export_config = ExporterConfig {
-        endpoint: otlp_endpoint,
-        ..ExporterConfig::default()
-    };
-    opentelemetry_otlp::new_metrics_pipeline(tokio::spawn, delayed_interval)
-        .with_export_config(export_config)
-        .with_aggregator_selector(selectors::simple::Selector::Histogram(vec![
-            0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9,
-        ]))
-        .with_period(Duration::from_secs_f64(period))
-        .build()
-}
-
-#[derive(Debug)]
-struct Config {
-    update_rate: f64,
-    push_rate: f64,
-    otlp_endpoint: String,
-}
-
-fn init_config() -> Result<Config, Box<dyn Error + Send + Sync + 'static>> {
-    let matches = App::new("obs_agent_client")
-        .arg(
-            Arg::with_name("update-rate")
-                .short("u")       
-                .long("update-rate")
-                .takes_value(true)
-                .help("Rate at which each instrument is updated with a new metric measurement (updates/sec)")
-        )
-        .arg(
-            Arg::with_name("push-rate")
-                .short("p")
-                .long("push-rate")
-                .takes_value(true)
-                .help("Rate at which measurements are pushed out of the client (pushes/sec)")
-        )
-        .arg(
-            Arg::with_name("otlp-endpoint")
-                .short("e")
-                .long("otlp-endpoint")
-                .takes_value(true)
-                .help("Endpoint to which OTLP messages will be sent.")
-        )
-        .get_matches();
-
-    let config = Config {
-        update_rate: std::env::var("UPDATE_RATE").map_or_else(
-            |_e| Ok(value_t!(matches.value_of("update-rate"), f64).unwrap_or(1.0)),
-            |v| v.parse(),
-        )?,
-        push_rate: std::env::var("PUSH_RATE").map_or_else(
-            |_e| Ok(value_t!(matches.value_of("push-rate"), f64).unwrap_or(0.2)),
-            |v| v.parse(),
-        )?,
-        otlp_endpoint: std::env::var("OTLP_ENDPOINT").map_or_else(
-            |_e| {
-                Ok(matches
-                    .value_of("otlp-endpoint")
-                    .unwrap_or("http://localhost:4317")
-                    .to_string())
-            },
-            |v| v.parse(),
-        )?,
-    };
-    Ok(config)
-}
+use obsagent_client::config;
+#[cfg(feature = "otel")]
+use obsagent_client::otel_client;
+#[cfg(feature = "prom")]
+use obsagent_client::prometheus_server;
 
 fn init_logging() {
     let subscriber = Subscriber::builder().with_max_level(Level::INFO).finish();
     let _ = subscriber::set_global_default(subscriber);
 }
 
 #[tokio::main]
-async fn main() -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
+async fn main() -> Result<(), anyhow::Error> {
```

Comment:
For types we don't use fully qualified name `anyhow:Error`. So just `Error` instead.

Reply:
If there is a naming conflict with another `Error` I believe we usually take anyhow without the fqn and then specify the other using an alias or fqn.

Reply:
Fixed.